client/post-editor/editor-sidebar/header.jsx

```diff
@@ -54,9 +54,8 @@ function EditorSidebarHeader( { translate, type, siteId, showDrafts, toggleDraft
 				</Button>
 			) }
 			{ type === 'post' && siteId && (
-				<PostCountsData siteId={ siteId } status="draft">
-					<DraftsButton onClick={ toggleDrafts } />
-				</PostCountsData>
+				<QueryPostCounts siteId={ siteId } type="post" /> ) && (
```

Comment:
Actually, now that I think of it, the query component should probably be in `<DraftsButton />` anyways (making it entirely self-sufficient). There, it could simply be a child of the `<Button />` return value.